include/rpc/function_proxy.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <string>
+#include <future>
+
+#include "msgpack.hpp"
+
+namespace rpc {
+
+class client;
+
+namespace call_type {
```

Comment:
Would it be possible for `get_function` to return a functor that knows if it is a sync or async function proxy?